tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+'''
+Utility script to read and write Prometheus Alert rules to file.
+Ensures we can test the Alert rules deployed and for documentation
+purposes.
+'''
+
+import argparse
+import ipaddress
+import json
+import pathlib
+import requests
+import subprocess
+import sys
+
+
+def query_prometheus_api(ip, port, route=None):
+    """Read the Prometheus route contents into memory."""
+    if not route:
+        sys.stderr.write("Expected endpoint route but got None")
+        sys.exit(1)
+
+    endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(ip, port, route)
+    try:
+        response = requests.get(url=endpoint, verify=False)
+    except requests.exceptions.ConnectionError as exc:
+        sys.stderr.write(
+            "Failed to reach endpoint {}: {!s}".format(route, exc)
+        )
+        sys.exit(1)
+
+    if response.status_code != 200:
+        sys.stderr.write(
+            "Expected status code <200> got {}".format(response.status_code)
+        )
+        sys.exit(1)
+
+    return response.json()
+
+
+def write_to_file(content, rule_name):
+    filename = '{}.json'.format(rule_name)
+    filepath = (pathlib.Path(__file__)/'..'/'..'/'..'/'..'
+            /'tools'/'rule_extractor'/filename).resolve()
+    with open('{}.json'.format(rule_name), 'w', encoding='utf-8') as f:
+        f.write(json.dumps(content, indent=4, sort_keys=True))
+
+
+def main():
+    """
+        python rule_extractor --ip 127.0.0.1 --port 9090 --type rules
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-i',
+        '--ip',
+        dest='ip',
+        help='Specify the target control-plane ingress IP address'
+    )
+    parser.add_argument(
+        '-p',
+        '--port',
+        dest='port',
+        default='8443',
+        help='Specify the target control-plane ingress port',
+        type=int
+    )
+    parser.add_argument(
+        '-t',
+        '--type',
+        required=True,
+        dest='object_type',
+        help='Specify the type of objects to fetch',
+        choices=['rules', 'alerts']
+    )
+
+    args = parser.parse_args()
+    if not args.ip:
+        # Obtain VM ip by connecting to it and running a command
+        # Requires that the VM is up and running.
+        ip = subprocess.check_output(
+            [
+                "vagrant",
+                "ssh",
+                "--command",
+                "hostname -I | cut -d' ' -f1"
+            ]
+        ).decode(sys.stdout.encoding).strip()
+        # Validate that we got a real ipv4 address
+        try:
+            ipaddress.ip_address(ip)
```

Comment:
No, this validate you got a valid IP address: it can be either v4 or v6.
- If we want to accept both, we should update the comment.
- If we only v4, the we should update the code (using `ipaddress.IPv4Address(ip)`)

Reply:
ok

Reply:
In fact, to be 100% safe you want `ip = str(ipaddress.IPv4Address(ip))` or check `ip == str(ipaddress.IPv4Address(ip))`

Because `IPv4Address` accepts `3232235521` as IP (which is technically valid), but I don't think Prometheus will be happy with that.